File: tools/ghidra_scripts/ExportStrings.py
# Ghidra post-analysis script: dump all defined strings to stdout
from ghidra.program.model.data import StringDataType, UnicodeDataType
from ghidra.program.model.listing import Data

listing = currentProgram.getListing()
data_iter = listing.getDefinedData(True)

count = 0
while data_iter.hasNext():
    d = data_iter.next()
    dt = d.getDataType()
    if isinstance(dt, (StringDataType, UnicodeDataType)):
        try:
            val = d.getValue()
            if val and len(str(val)) > 3:
                addr = d.getAddress()
                print("0x{}\t{}".format(addr, repr(str(val))))
                count += 1
        except Exception:
            pass

print("# Total strings: {}".format(count), file=java.lang.System.err)

File: tools/ghidra_scripts/ExportStrings.java
//Export all defined strings (length > 3) with address to stdout
//@category EawMT

import ghidra.app.script.GhidraScript;
import ghidra.program.model.listing.*;
import ghidra.program.model.data.*;
import ghidra.program.model.address.*;

public class ExportStrings extends GhidraScript {
    @Override
    public void run() throws Exception {
        Listing listing = currentProgram.getListing();
        DataIterator it = listing.getDefinedData(true);
        java.io.PrintWriter pw = new java.io.PrintWriter(
            new java.io.FileWriter("/home/ty/src/eaw-reverse-engineering/logs/ghidra_strings.tsv"));
        int count = 0;
        while (it.hasNext()) {
            Data d = it.next();
            DataType dt = d.getDataType();
            if (dt instanceof StringDataType || dt instanceof UnicodeDataType ||
                dt instanceof Unicode32DataType || dt instanceof TerminatedStringDataType ||
                dt instanceof TerminatedUnicodeDataType) {
                try {
                    Object val = d.getValue();
                    if (val != null) {
                        String s = val.toString();
                        if (s.length() > 3) {
                            pw.println(d.getAddress() + "\t" + s);
                            count++;
                        }
                    }
                } catch (Exception e) { /* skip */ }
            }
        }
        pw.close();
        println("Wrote " + count + " strings to logs/ghidra_strings.tsv");
    }
}
